packages/fxa-settings/src/pages/Signin/container.tsx

```diff
@@ -0,0 +1,294 @@
+    hasLinkedAccount: hasLinkedAccountFromLocationState,
+    hasPassword: hasPasswordFromLocationState,
+  } = location.state || {};
+
+  const [accountStatus, setAccountStatus] = useState({
+    hasLinkedAccount:
+      queryParamModel.hasLinkedAccount || hasLinkedAccountFromLocationState,
+    hasPassword: queryParamModel.hasPassword || hasPasswordFromLocationState,
+  });
+  const { hasLinkedAccount, hasPassword } = accountStatus;
+
+  const nonCachedEmail = queryParamModel.email || emailFromLocationState;
+  let email = nonCachedEmail;
+  let sessionToken: hexstring | undefined;
+  let uid: hexstring | undefined;
+  // only read from local storage if email isn't provided via query param or router state
+  if (!nonCachedEmail) {
+    const storedLocalAccount = currentAccount();
+    email = storedLocalAccount?.email;
+    sessionToken = storedLocalAccount?.sessionToken;
+    uid = storedLocalAccount?.uid;
+  }
+
+  const isOAuth = isOAuthIntegration(integration);
+  const isSyncOAuth = isOAuth && integration.isSync();
+  const isSyncDesktopV3 = isSyncDesktopV3Integration(integration);
+  const isSyncWebChannel = isSyncOAuth || isSyncDesktopV3;
+
+  useEffect(() => {
+    (async () => {
+      // Tweak this once index page is converted to React
+      if (!validationError && email) {
+        // if you directly hit /signin with email param or we read from localstorage
+        // this means the account status hasn't been checked
+        if (
+          accountStatus.hasLinkedAccount === undefined ||
+          accountStatus.hasPassword === undefined
+        ) {
+          // TODO: error handling for this (and in SignUp)
+          const { exists, hasLinkedAccount, hasPassword } =
+            await authClient.accountStatusByEmail(email, {
+              thirdPartyAuthStatus: true,
+            });
+          if (!exists) {
+            // For now, just pass back emailStatusChecked. When we convert the Index page
+            // we'll want to read from router state.
+            navigate(`/signup?email=${email}&emailStatusChecked=true`);
+            // TODO: Probably move this to the Index page onsubmit once
+            // the index page is converted to React, we need to run it in
+            // signup and signin for Sync
+          } else {
+            setAccountStatus({
+              hasLinkedAccount,
+              hasPassword,
+            });
+            if (isSyncWebChannel) {
+              firefox.fxaCanLinkAccount({ email: queryParamModel.email });
+            }
+          }
+        } else if (isSyncWebChannel) {
+          // TODO: Probably move this to the Index page onsubmit once
+          // the index page is converted to React, we need to run it in
+          // signup and signin for Sync
+          firefox.fxaCanLinkAccount({ email: queryParamModel.email });
+        }
+      }
+    })();
+  });
+
+  const { data: avatarData, loading: avatarLoading } =
+    useQuery<AvatarResponse>(AVATAR_QUERY);
```

Comment:
I believe this is how Backbone does it, I used to wonder why it sometimes didn't show the avatar until working on this. This just shows the default in this case which seems okay / to parity.

Reply:
Potentially, if https://github.com/mozilla/fxa/pull/16279 is merged in first, we could verify if the session is verified first, and if not skip the avatar query to avoid the must verify error? 🤔

Reply:
Nice, I'll look at this in a follow up.